Bio/GenBank/__init__.py

```diff
@@ -322,7 +324,8 @@ def _loc(loc_str, expected_seq_length, strand, is_circular=False):
     try:
         s, e = loc_str.split("..")
     except ValueError:
-        assert ".." not in loc_str
+        if ".." in loc_str:
+            raise ValueError("Expected '..' to be absent from 'loc_str'.")
```

Comment:
Again, avoid using the local variable name in the exception message if possible ('loc_str'). How about:

``ValueError(f"Unexpected use of '..' in location '{loc_str}'")``

This branch of the code would be triggered by either no ".." (valid but rare), or multiple cases which is what the assert is really guarding against.